dnsdist: signed/unsigned comparison warning

- Program: dnsdist
 - Issue type: Bug report

### Short description
it looks like the TimedIPSetRule does signed/unsigned comparisons:

```
In file included from dnsdist-lua.cc:26:0:
dnsrulactions.hh: In member function ‘virtual bool TimedIPSetRule::matches(const DNSQuestion*) const’:
dnsrulactions.hh:167:22: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
       return time(0) < fnd->second;
                      ^
dnsrulactions.hh:174:22: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
       return time(0) < fnd->second;
                      ^
dnsrulactions.hh: In member function ‘void TimedIPSetRule::cleanup()’:
dnsrulactions.hh:224:18: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
  if(iter->second < now)
                  ^
dnsrulactions.hh:236:18: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
  if(iter->second < now)
                  ^
dnsrulactions.hh: In member function ‘virtual std::string TimedIPSetRule::toString() const’:
dnsrulactions.hh:253:16: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
         if(now < ip.second)
                ^
dnsrulactions.hh:259:16: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
         if(now < ip.second)
                ^
```

Found when building dnsdist on raspbian (ARM)

Another one:

```
dnsdist-tcp.cc: In function ‘bool maxConnectionDurationReached(unsigned int, time_t, unsigned int&)’:
dnsdist-tcp.cc:194:17: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     if (elapsed >= maxConnectionDuration) {
                 ^
```

--- a/pdns/dnsdist-tcp.cc
+++ b/pdns/dnsdist-tcp.cc
@@ -190,7 +190,11 @@ static bool sendResponseToClient(int fd, const char* response, uint16_t response
 static bool maxConnectionDurationReached(unsigned int maxConnectionDuration, time_t start, unsigned int& remainingTime)
 {
   if (maxConnectionDuration) {
-    time_t elapsed = time(NULL) - start;
+    time_t curtime = time(nullptr);
+    unsigned int elapsed = 0;
+    if (curtime > start) { // To prevent issues when time goes backward
+      elapsed = curtime - start;
+    }
     if (elapsed >= maxConnectionDuration) {
       return true;
     }

--- a/pdns/dnsrulactions.hh
+++ b/pdns/dnsrulactions.hh
@@ -272,8 +272,8 @@ private:
       return ah & (bh<<1);
     }
   };
-  std::unordered_map<IPv6, uint32_t, IPv6Hash> d_ip6s;
-  std::unordered_map<uint32_t, uint32_t> d_ip4s;
+  std::unordered_map<IPv6, time_t, IPv6Hash> d_ip6s;
+  std::unordered_map<uint32_t, time_t> d_ip4s;
   mutable pthread_rwlock_t d_lock4;
   mutable pthread_rwlock_t d_lock6;
 };

